### PR TITLE
Log version in lighthouse pods

### LINKED ISSUE #1279: Lighthouse components should log their version on startup
**What would you like to be added**:

Each component (each `main.go` file) should log its version on startup, [in the same way as the operator](https://github.com/submariner-io/submariner-operator/blob/8fb881cb70d8ffe7eb04545c7a4071af895e7bb5/main.go#L69) (with a version defined at build-time).

**Why is this needed**:

This will allow deployed versions to be determined precisely.

### PR DESCRIPTION
**Pull request by vthapar**
Log version for Lighthoue agent and coredns pods on startup.

Fixes #1279

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,6 +25,7 @@ TARGETS := $(shell ls -p scripts | grep -v -e / -e deploy)
 override E2E_ARGS += cluster1 cluster2 cluster3
 override UNIT_TEST_ARGS += test/e2e
 export LIGHTHOUSE = true
+export LDFLAGS = -X main.VERSION=$(VERSION)
```
Comment (skitt) [inline Makefile]:
```suggestion
export LDFLAGS = -X main.version=$(VERSION)
```

File: coredns/main.go [hunk 2]
```diff
@@ -105,7 +105,10 @@ var directives = []string{
 	"on",
 }
 
+var VERSION = "not-compiled-properly"
```
Comment (skitt) [inline coredns/main.go]:
```suggestion
var version = "not-compiled-properly"
```

File: coredns/main.go [hunk 3]
```diff
@@ -105,7 +105,10 @@ var directives = []string{
...
+
 func init() {
+	lighthouse.VERSION = VERSION
```
Comment (skitt) [inline coredns/main.go]:
```suggestion
	lighthouse.Version = version
```

File: coredns/plugin/setup.go [hunk 4]
```diff
@@ -42,6 +42,7 @@ import (
 var (
 	masterURL  string
 	kubeconfig string
+	VERSION    string
```
Comment (skitt) [inline coredns/plugin/setup.go]:
```suggestion
	Version    string
```

File: coredns/plugin/setup.go [hunk 5]
```diff
@@ -70,7 +71,7 @@ func init() {
 // setup is the function that gets called when the config parser see the token "lighthouse". Setup is responsible
 // for parsing any extra options the this plugin may have. The first token this function sees is "lighthouse".
 func setupLighthouse(c *caddy.Controller) error {
-	log.Debug("In setupLighthouse")
+	log.Infof("submariner-lighthouse-coredns version: %v", VERSION)
```
Comment (skitt) [inline coredns/plugin/setup.go]:
```suggestion
	log.Infof("submariner-lighthouse-coredns version: %v", Version)
```

File: pkg/agent/main.go [hunk 6]
```diff
@@ -48,6 +48,7 @@ var (
 	kubeConfig string
 	help       = false
 	logger     = log.Logger{Logger: logf.Log.WithName("main")}
+	VERSION    = "not-compiled-properly"
```
Comment (skitt) [inline pkg/agent/main.go]:
```suggestion
	version    = "not-compiled-properly"
```

File: pkg/agent/main.go [hunk 7]
```diff
@@ -111,7 +112,7 @@ func main() {
 	localClient, err := dynamic.NewForConfig(cfg)
 	exitOnError(err, "Error creating dynamic client")
 
-	logger.Info("Starting submariner-lighthouse-agent")
+	logger.Infof("Starting submariner-lighthouse-agent version: %v", VERSION)
```
Comment (skitt) [inline pkg/agent/main.go]:
```suggestion
	logger.Infof("Starting submariner-lighthouse-agent version: %v", version)
```

**Review by skitt — CHANGES_REQUESTED** [review]

File: coredns/main.go (hunk 3 above)
Comment (tpantelis) [reply coredns/main.go]: Why don't we just do `log.Infof("submariner-lighthouse-coredns version: %v", version)` in `main` before L116 below?

File: pkg/agent/main.go [hunk 8]
```diff
@@ -111,7 +112,7 @@ func main() {
 	localClient, err := dynamic.NewForConfig(cfg)
 	exitOnError(err, "Error creating dynamic client")
 
-	logger.Info("Starting submariner-lighthouse-agent")
+	logger.Infof("Starting submariner-lighthouse-agent version: %v", version)
```
Comment (skitt) [inline pkg/agent/main.go]: Grammatically, this should be

```suggestion
	logger.Infof("Starting submariner-lighthouse-agent version %v", version)
```

but to keep it consistent with other version output, and perhaps simplify parsing should we want to extract this in `subctl`, let’s make it

```suggestion
	logger.Infof("submariner-lighthouse-agent version: %v", version)
```

**Review by skitt — DISMISSED** [review]
Sorry, something else I didn’t notice earlier.

**Review by skitt — APPROVED** [review]
In future, please don’t dismiss “stale” reviews when you address the changes requested — you can mark the changes resolved, but dismissing reviews should only be done if necessary to unblock a PR.

**Review by aswinsuryan — APPROVED** [review]

**Review by tpantelis — APPROVED** [review]

**Pull request merged**